src/inverted-index.js

```diff
@@ -20,96 +21,89 @@ class InvertedIndex {
    *
    * @param {String} document
    * @param {Object} file
+   * @return {Bool}
    */
   createIndex(fileName, file) {
 
     file = this._verifyFile(file);
-    if (!file) return false;
+    if (!file) { return false; }
 
     const words = []
-        , indexed = {};
+        , indexedWords = {};
 
-    file.forEach((eachFileDoc) => {
-      words.push(this._tokenize(this._stringify(eachFileDoc)));
+    file.forEach((document) => {
+      words.push(this._tokenize(this._stringify(document)));
     });
 
-    words.forEach((eachWordsArray, index) => {
-      for (let i = 0; i < eachWordsArray.length; i++) {
-        if (!indexed.hasOwnProperty(eachWordsArray[i])) {
-          indexed[eachWordsArray[i]] = [];
+    words.forEach((documentWords, index) => {
+      for (let i = 0; i < documentWords.length; i++) {
+        if (!indexedWords.hasOwnProperty(documentWords[i])) {
+          indexedWords[documentWords[i]] = [];
         }
-        indexed[eachWordsArray[i]].push(index);
+        indexedWords[documentWords[i]].push(index);
       }
     });
 
-    if (!this.indeces.hasOwnProperty(fileName)) this.indeces[fileName] = indexed;
-    this.indexed[fileName] = file;
+    if (!this.indices.hasOwnProperty(fileName)) {
+      this.indices[fileName] = indexedWords;
+    }
+
+    this.indexedFiles[fileName] = file.length;
     return true;
   }
 
   /**
-   * getIndex method returns the indeces object
+   * getIndex method returns the indices object
    *
    * @param {String} fileName
-   * @return {Object} indeces
+   * @return {Object} indices
    */
   getIndex(fileName) {
-    try {
-      return typeof fileName == 'undefined'
-        || typeof fileName !== 'string' ? this.indeces : this.indeces[fileName];
-    } catch(e) {
-      throw new Error('something went wrong', e);
+    if (Object.keys(this.indices).length === 0) {
+      return false;
     }
+
+    return typeof fileName === 'undefined'
+        || typeof fileName !== 'string' ? this.indices : this.indices[fileName];
   }
 
   /**
    * searchIndex method looks up searchTerm
-   * in the document and returns an indeces
+   * in the document and returns an indices
    *
    * @param {String} document
    * @param {String} searchTerm
    * @return {Object} result
    */
-  searchIndex(file, searchTerm) {
+  searchIndex(fileName, searchTerm) {
+
     let result = {}
       , termsArr = [];
 
+    if (Array.isArray(searchTerm)) {
+      searchTerm = searchTerm.join(',').split(',').join(' ');
```

Comment:
Is `searchTerm.join(',').split(',')` not going to be equal to `searchTerm`? If so then it's not necessary. You can just do `searchTerm`.join(' ')`